src/Microsoft.Identity.Client/Azure/Constants.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        public const string AzureCertificateStoreLocation = "AZURE_CERTIFICATE_STORE_LOCATION";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on AppService
+        /// </summary>
+        public const string ManagedIdentityAppServiceApiVersion = "2017-08-01";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on Virtual Machines
+        /// </summary>
+        public const string ManagedIdentityVMApiVersion = "2018-02-01";
+
+        /// <summary>
+        /// Managed Identity loopback IP address
+        /// </summary>
+        public const string ManagedIdentityLoopbackAddress = "169.254.169.254";
+
+        /// <summary>
+        /// Managed Identity metadata endpoint
+        /// </summary>
+        public const string ManagedIdentityMetadataEndpoint = "http://" + ManagedIdentityLoopbackAddress + "/metadata";
+
+        /// <summary>
+        /// Managed Identity 
+        /// </summary>
+        public const string ManagedIdentityTokenEndpoint = ManagedIdentityMetadataEndpoint + "/identity/oauth2/token";
+
+        /// <summary>
+        /// ManagedIdentityEndpointEnvName is the name of the environment variable which holds the endpoint for the
+        /// managed identity service running in Azure AppService 
+        /// </summary>
+        public const string ManagedIdentityEndpointEnvName = "MSI_ENDPOINT";
+
+        /// <summary>
+        /// ManagedIdentitySecretEnvName is the name of the environment variable which holds the secret for use in
+        /// calling the managed identity service running in Azure AppService
+        /// </summary>
+        public const string ManagedIdentitySecretEnvName = "MSI_SECRET";
+
+        /// <summary>
+        /// AadAuthorityEnvName is the name of the environment variable which holds the hostname of the service token service (STS) from which MSAL.NET will acquire the tokens. Ex. login.microsoftonline.com
+        /// </summary>
+        public const string AadAuthorityEnvName = "AAD_AUTHORITY";
+    }
+
+    /// <summary>
+    /// ErrorCodes contains string error codes that can be emitted from this namespace
+    /// </summary>
+    public static class ErrorCodes
+    {
+        /// <summary>
+        /// FailedParseOfManagedIdentityExpiration happens when we can't parse the expiry on the token
+        /// </summary>
+        public const string FailedParseOfManagedIdentityExpiration = "failed_parse_of_managed_identity_token_expiry";
+
+        public const string NoResourceUriInScopes = "no_resource_uri_with_slash_.default_in_scopes";
+
+        public const string NoProbesAvailable = "no_probes_are_available";
+
+        public const string MaxRetriesExhausted = "max_retries_exhausted";
+    }
+
+    /// <summary>
+    ///     Env provides access to well known environment variables
+    /// </summary>
+    public static class Env
```

Comment:
I'd like to discuss:
- bringing some of these with the `ConfidentialClientApplicationOptions`. A lot are there already, but not what is about the certificates. This would be a great addition to provide a certificate from the config file or other was
- providing a layer which leverages the [Configuration in .NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2) which maps [environement variables](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.2#environment-variables-configuration-provider), command line flags, sections in config files etc ... to a POCO structure

Reply:
Great idea!